main/src/cgeo/geocaching/gcvote/GCVote.java

```diff
@@ -42,6 +43,64 @@ private GCVote() {
         // utility class
     }
 
+    private static class SingletonHolder {
+        private final static GCVote INSTANCE = new GCVote();
+    }
+
+    public static GCVote getInstance() {
+        return SingletonHolder.INSTANCE;
+    }
+
+    public static StatusCode login() {
+        final ImmutablePair<String, String> login = Settings.getGCVoteLogin();
+
+        if (login == null) {
+            Log.e("Credentials can't be retreived");
+            return StatusCode.NO_LOGIN_INFO_STORED;
+        }
+
+        if (StringUtils.isEmpty(login.left) || StringUtils.isEmpty(login.right)) {
+            Log.e("GCVote login information is not stored");
+            return StatusCode.NO_LOGIN_INFO_STORED;
+
+        }
+
+        final Parameters params = new Parameters("version", "cgeo");
+        params.put("userName", login.left, "password", login.right);
+
+        final InputStream response = Network.getResponseStream(Network.getRequest("http://gcvote.com/getVotes.php", params));
+        boolean loggedIn = false;
```

Comment:
Why this extra variable instead of just returning prematurely from the `while` loop?